src/main/java/me/youhavetrouble/purpurextras/listeners/SpawnerPlacementListener.java

```diff
@@ -0,0 +1,23 @@
+package me.youhavetrouble.purpurextras.listeners;
+
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.format.NamedTextColor;
+import org.bukkit.Material;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.BlockPlaceEvent;
+
+public class SpawnerPlacementListener implements Listener {
+
+    @EventHandler
+    public void onSpawnerPlace(BlockPlaceEvent event){
+        Material block = event.getBlock().getType();
+        Player player = event.getPlayer();
+        if (!(block.equals(Material.SPAWNER))) return;
+        if (!(player.hasPermission("purpurextras.spawnerPlace"))){
```

Comment:
permissions should always be all lowercase.

Also, I believe there's a room for improvement here, why not check spawner's entity type and check for permission specific to that entity? Could add wildcard permission for all of them

Reply:
I did try that, actually, but couldn't figure out how to get it to work